### PR TITLE
Basp-MR-35: SetUp Enviroment

### PR DESCRIPTION
**Pull request by josedaniele**
This PR includes the initial setup of the development environment, which involves installing dependencies and configuring them. Additionally, it creates the necessary folders for tests and seeds, along with their respective files, to kickstart the project

### CODE REVIEW

**Review by JuanMornik — APPROVED** [review]

File: .husky/pre-commit [hunk 1]
```diff
@@ -2,3 +2,4 @@
 . "$(dirname "$0")/_/husky.sh"
 
 npm run lint
```
Comment (NicoCagnina) [inline .husky/pre-commit:4]: simply put `npm run lint && npm run test`

File: jest.setup.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    useUnifiedTopology: true,
+  };
+  await mongoose.connect(uri, mongoOptions);
+});
+
+// Disconnect from the fake database
+afterAll(async () => {
+  await mongoose.connection.dropDatabase();
+  await mongoose.connection.close();
+  await mongoServer.stop();
```
Comment (NicoCagnina) [inline jest.setup.js:23]: we should add `mongo.stop()` here

File: src/tests/activity.test.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// Test basico para evitar conflictos.
```
Comment (NicoCagnina) [inline src/tests/activity.test.js:1]: english please

File: src/index.js [hunk 4]
```diff
@@ -1,24 +1,22 @@
...
-mongoose
-  .connect(DB_URL, { maxPoolSize: process.env.MONGO_POOLSIZE || 1})
-  .then(() => console.log('Db connected'))
-  .catch((error) => console.log('Error : ', error));
-
-app.use('/api', router);
-
-app.listen(port, () => {
-  console.log(`Example app listening on port ${port}`);
+app.get('/', (req, res) => {
```
Comment (NicoCagnina) [inline src/index.js:11]: why is this here?

File: src/index.js [hunk 5]
```diff
@@ -1,24 +1,22 @@
...
+const dontenv = require('dotenv');
 const mongoose = require('mongoose');
+const app = require('./app');
 
-const router = require('./routes');
+dontenv.config();
 
-const app = express();
-const port = process.env.PORT || 4000;
+mongoose.connect(process.env.MONGO_DB_CONNECT_URL)
```
Comment (NicoCagnina) [inline src/index.js:7]: this should be: 

```
mongoose.connect(
process.env.MONGO_DB_CONNECT_URL, 
{ maxPoolSize: process.env.MONGO_POOLSIZE || 1})
```

File: src/tests/activity.test.js (hunk 3 above)
Comment (NicoCagnina) [reply src/tests/activity.test.js:1]: for ex:
```
// Basic test to avoid conflicts
// REMOVE THIS TEST AND MAKE YOURS!
test('should pass a basic test', () => {
  expect(1 + 1).toBe(2);
});
```

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

**Review by NicoCagnina — CHANGES_REQUESTED** [review]
You should add the .env file in the .gitignore. Otherwise, all your colleagues will upload the .env file.

File: src/index.js (hunk 4 above)
Comment (josedaniele) [reply src/index.js:11]: Idk

File: src/index.js (hunk 5 above)
Comment (josedaniele) [reply src/index.js:7]: they didn't explain that 😡

**Comment by josedaniele** [comment]
> You should add the .env file in the .gitignore. Otherwise, all your colleagues will upload the .env file.

this is already inside the file

File: src/index.js (hunk 5 above)
Comment (NicoCagnina) [reply src/index.js]: it was on the file before you updated it.

**Review by NicoCagnina — APPROVED** [review]
